### PR TITLE
fixes #374, updated docs, fixed e2e tests

### LINKED ISSUE #374: Update Adobe dependencies in order to incorporate HTTP proxy fix
**Expected Behavior**

This library should be up to date, i.e. (transitively) incorporate fixes made in dependencies

**Actual Behavior**

The library is outdated. It references an old version of @adobe/aio-lib-core-networking and hence the http proxy logic is faulty (see https://github.com/adobe/aio-lib-core-networking/issues/16). 

**Reproduction Scenario, Platform, and Version**

Use the aio cli cloudmanager plugin (https://github.com/adobe/aio-cli-plugin-cloudmanager) with an http proxy. It won't be able to connect with the cloud manager API, since the proxy configuration is wrong (an http-proxy agent will be used where an https-proxy-agent should be used). 

**Steps to Reproduce**

1. Setup an http proxy (e.g. use squid)
2. Run the aio cli cloudmanager plugin e2e tests with an HTTP proxy configured
3. Calls to the Cloud Manager API will error out

**Steps to Fix**

1. Create a new version of this library
2. Bump up this library's dependency @adobe/aio-lib-core-networking to version 3.0.0
3. Link the newly created version to aio-cli-cloudmanager
4. And it will work
5. While at it, fix this library's e2e tests

### PR DESCRIPTION
**Pull request by pahupe**
BREAKING CHANGE: dependent library reference has been updated to a new major version

## Description

Fix for #374 : Updated dependencies
While at it, I repaired the e2e tests and updated the docs accordingly: replaced doSomething() with existing API call listPrograms() and replaced "tenantId" with "orgId"

## Related Issue

#374 

## Motivation and Context

aio cli cloudmanager (the plugin) does not work with http proxies and this is a first step to fix it, as the plugin depends on this library

## How Has This Been Tested?

Locally using e2e tests and e2e tests with a proxy configuration

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change contains a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have fixed tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: e2e/e2e.js [hunk 1]
```diff
@@ -16,46 +16,46 @@ const path = require('path')
 require('dotenv').config({ path: path.join(__dirname, '.env') })
 
 let sdkClient = {}
-const tenantId = process.env['{{LIB_NAME}}_TENANT_ID']
+const orgId = process.env['{{LIB_NAME}}_TENANT_ID']
```
Comment (moritzraho) [inline e2e/e2e.js:19]: Why is the env still called TENANT_ID ?

Comment (pahupe) [reply e2e/e2e.js:19]: Good point, I can change it to ORG_ID as well. The issue that was fixed here is: the internal field is now called sdkClient.orgId (see line 28: expect(sdkClient.orgId).toBe(orgId))

Comment (pahupe) [reply e2e/e2e.js]: @moritzraho : Added a commit where I renamed all occurrences from tenantId to orgId

**Review by moritzraho — APPROVED** [review]

File: e2e/e2e.js (hunk 1 above)
Comment (moritzraho) [reply e2e/e2e.js]: ok thanks

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@adobe/aio-lib-cloudmanager",
-  "version": "2.0.0",
+  "version": "3.0.0",
```
Comment (mbradoschi) [inline package.json:3]: If I'm not mistaken, this needs to be handled by `semantic-release` bot.

Considering https://github.com/adobe/aio-lib-core-networking/issues/16 is a breaking change, I agree we should have this update in a major version, but I've looked through https://github.com/adobe/aio-lib-cloudmanager/blob/main/.github/CONTRIBUTING.md#commits-and-releasing and it seems we need to append `BREAKING CHANGE: <description>` into our commits for the `semantic-release-bot` to create a `3.0.0`.

Here's an example that I've found: #359.

Comment (pahupe) [reply package.json:3]: Ok, thanks. Let me revert that change and add a BREAKING CHANGE line to the commit

**Review by mbradoschi — APPROVED** [review]
Changes look good to me.

I've tested `aio-cli-plugin-cloudmanager` locally with this version of `aio-lib-cloudmanager` and I haven't found any issues.

**Pull request merged**